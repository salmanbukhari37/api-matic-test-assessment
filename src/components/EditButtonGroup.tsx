import React from "react";
import { Box, Button } from "@mui/material";
import { EditButtonGroupProps } from "dto/interfaces/IEditButtonGroup.interface";

const EditButtonGroup: React.FC<EditButtonGroupProps> = ({
  isEditing,
  onSave,
  onEdit,
  onCancel,
}) => {
  return (
    <Box position="absolute" top="16px" right="16px" display="flex" gap={1}>
      {isEditing ? (
        <>
          <Button
            variant="contained"
            sx={{
              backgroundColor: "#28a745",
              color: "#fff",
              textTransform: "capitalize",
              fontWeight: "bold",
              "&:hover": {
                backgroundColor: "#218838",
              },
            }}
            onClick={onSave}
          >
            Complete Edit
          </Button>
          <Button
            variant="contained"
            sx={{
              backgroundColor: "#dc3545",
              color: "#fff",
              textTransform: "capitalize",
              fontWeight: "bold",
              "&:hover": {
                backgroundColor: "#c82333",
              },
            }}
            onClick={onCancel}
          >
            Cancel
          </Button>
        </>
      ) : (
        <Button
          variant="contained"
          sx={{
            backgroundColor: "#007bff",
            color: "#fff",
            textTransform: "capitalize",
            fontWeight: "bold",
            "&:hover": {
              backgroundColor: "#0056b3",
            },
          }}
          onClick={onEdit}
        >
          Edit
        </Button>
      )}
    </Box>
  );
};

export default EditButtonGroup;
