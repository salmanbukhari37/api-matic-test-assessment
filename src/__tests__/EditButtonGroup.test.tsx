import { render, screen, fireEvent } from "@testing-library/react";
import EditButtonGroup from "../components/EditButtonGroup";

describe("EditButtonGroup Component", () => {
  const mockOnSave = jest.fn();
  const mockOnEdit = jest.fn();
  const mockOnCancel = jest.fn();

  const defaultProps = {
    isEditing: false,
    onSave: mockOnSave,
    onEdit: mockOnEdit,
    onCancel: mockOnCancel,
  };

  test("matches snapshot when not editing", () => {
    const { container } = render(
      <EditButtonGroup {...defaultProps} isEditing={false} />
    );
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot when editing", () => {
    const { container } = render(
      <EditButtonGroup {...defaultProps} isEditing={true} />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders Edit button when not editing", () => {
    render(<EditButtonGroup {...defaultProps} />);

    const editButton = screen.getByRole("button", { name: /edit/i });
    expect(editButton).toBeInTheDocument();

    fireEvent.click(editButton);
    expect(mockOnEdit).toHaveBeenCalledTimes(1);
  });

  test("renders Complete Edit and Cancel buttons when editing", () => {
    render(<EditButtonGroup {...defaultProps} isEditing={true} />);

    const completeEditButton = screen.getByRole("button", {
      name: /complete edit/i,
    });
    const cancelButton = screen.getByRole("button", { name: /cancel/i });

    expect(completeEditButton).toBeInTheDocument();
    expect(cancelButton).toBeInTheDocument();

    fireEvent.click(completeEditButton);
    expect(mockOnSave).toHaveBeenCalledTimes(1);

    fireEvent.click(cancelButton);
    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  test("does not call any handlers without interaction", () => {
    render(<EditButtonGroup {...defaultProps} isEditing={true} />);

    expect(mockOnSave).not.toHaveBeenCalled();
    expect(mockOnCancel).not.toHaveBeenCalled();
    expect(mockOnEdit).not.toHaveBeenCalled();
  });
});
