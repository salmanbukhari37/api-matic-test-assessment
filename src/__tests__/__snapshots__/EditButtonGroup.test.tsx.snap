// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditButtonGroup Component matches snapshot when editing 1`] = `
<div>
  <div
    class="MuiBox-root css-gj66ts"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1b34cgk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Complete Edit
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-kx3slk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`EditButtonGroup Component matches snapshot when not editing 1`] = `
<div>
  <div
    class="MuiBox-root css-gj66ts"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1nkrjv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Edit
    </button>
  </div>
</div>
`;
